### PR TITLE
Update zboss libraries to v3.5.2.0

### PR DESCRIPTION
**Pull request by tomchy**
* Providing the two sets of libraries allowed us to add experimental features with the production version intact.
After the NCS v1.5.1 release, there will be only a single library to reduce the complexity of the Zigbee stack configuration.
* Update ZBOSS libraries to v3.5.2.0

### CODE REVIEW

**Review by greg-fer — CHANGES_REQUESTED** [review]
Zigbee memory requirements page also needs to be updated, as it contains tables for development and production libraries.

File: doc/nrf/ug_zigbee_configuring.rst [hunk 1]
```diff
@@ -20,11 +20,6 @@ Zigbee requires the following modules to properly operate in |NCS|:
   OSIF implements a series of functions used by ZBOSS and is included in the |NCS|'s Zigbee subsystem.
   The files that handle the OSIF integration are located in :file:`nrf/subsys/zigbee/osif`.
```
[the remark this answers is not in the document]
Comment (tomchy) [reply doc/nrf/ug_zigbee_configuring.rst:22]: Fixed.

File: samples/zigbee/ncp/README.rst [hunk 2]
```diff
@@ -49,7 +49,7 @@ Overview
 
 The sample demonstrates using a Nordic Semiconductor's Development Kit as Zigbee Network Co-Processor.
 
-The sample uses the :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` Kconfig option, which is available in the :ref:`development version of the ZBOSS stack libraries <nrfxlib:zboss_configuration>` (managed by the :option:`CONFIG_ZIGBEE_LIBRARY_DEVELOPMENT` Kconfig option, also enabled by default).
+The sample uses the :option:`CONFIG_ZIGBEE_LIBRARY_NCP_DEV` Kconfig option, which is available in the :ref:`ZBOSS stack libraries <nrfxlib:zboss_configuration>`.
```
[the remark this answers is not in the document]
Comment (tomchy) [reply samples/zigbee/ncp/README.rst:52]: Fixed

File: subsys/zigbee/Kconfig [hunk 3]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
...
+	  - 0x00000020 Zigbee Smart Energy
+	  - 0x00000010 Application Support layer
+	  - 0x00000008 Network layer
+	  - 0x00000004 MAC layer
+	  - 0x00000002 Memory management
+	  - 0x00000001 Common
+
+	  For example, in order to enable traces related to OTA DFU, one should set this option to 0x00080100.
+
+	  Note: For general debugging purposes, please use 0x00000C48.
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:196]: Do not use `please`.

File: subsys/zigbee/Kconfig [hunk 4]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
...
+	  - 0x00000080 Security
+	  - 0x00000040 Zigbee Device Object
+	  - 0x00000020 Zigbee Smart Energy
+	  - 0x00000010 Application Support layer
+	  - 0x00000008 Network layer
+	  - 0x00000004 MAC layer
+	  - 0x00000002 Memory management
+	  - 0x00000001 Common
+
+	  For example, in order to enable traces related to OTA DFU, one should set this option to 0x00080100.
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:194]:
```suggestion
	  For example, to enable traces related to OTA DFU, set this option to 0x00080100.
```

File: subsys/zigbee/Kconfig [hunk 5]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
 config ZBOSS_TRACE_MASK
 	hex "Trace mask of ZBOSS stack logs"
 	default 0x0000
-	range 0x0000 0xFFFF
+	range 0x00000000 0xFFFFFFFF
 	help
 	  Selectively enable Zigbee binary trace logs.
 	  The mask value should be a bitwise OR of values assigned to selected modules.
 
 	  Available modules are:
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:164]:
```suggestion
	  Available modules:
```

File: subsys/zigbee/Kconfig [hunk 6]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
 config ZBOSS_TRACE_MASK
 	hex "Trace mask of ZBOSS stack logs"
 	default 0x0000
-	range 0x0000 0xFFFF
+	range 0x00000000 0xFFFFFFFF
 	help
 	  Selectively enable Zigbee binary trace logs.
 	  The mask value should be a bitwise OR of values assigned to selected modules.
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:162]:
```suggestion
	  The mask value is expected to be a bitwise OR of values assigned to selected modules.
```

File: subsys/zigbee/Kconfig [hunk 7]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
...
-
-	  Note: For general debugging purposes, please use 0x0C48.
+	  - 0x04000000 Reserved
+	  - 0x02000000 Reserved
+	  - 0x01000000 Reserved
+	  - 0x00800000 Reserved
+	  - 0x00400000 Reserved
+	  - 0x00200000 Zigbee NCP: transport (LL)
+	  - 0x00100000 Zigbee NCP: command adapters (HL)
+	  - 0x00080000 ZCL: Over The Air firmware upgrade
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:173]: Shouldn't this be FOTA?
`ZCL: Firmware over-the-air upgrade`?

File: subsys/zigbee/Kconfig [hunk 8]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
...
+	  - 0x00040000 IAS zone device: battery monitoring
+	  - 0x00020000 Reserved
+	  - 0x00010000 MAC Lower Layer
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:176]: `MAC lower layer`?

File: subsys/zigbee/Kconfig [hunk 9]
```diff
@@ -156,33 +156,44 @@ config ZBOSS_TRACE_LOG_LEVEL
...
+	  - 0x00002000 Custom components
+	  - 0x00001000 Reserved
+	  - 0x00000800 Application
+	  - 0x00000400 Reserved
+	  - 0x00000200 Zigbee Light Link
+	  - 0x00000100 Zigbee Cluster Library
+	  - 0x00000080 Security
+	  - 0x00000040 Zigbee Device Object
+	  - 0x00000020 Zigbee Smart Energy
+	  - 0x00000010 Application Support layer
```
Comment (greg-fer) [inline subsys/zigbee/Kconfig:188]: `Application support layer`?

**Review by greg-fer — APPROVED** [review]
Thanks for addressing my comments.
Approving with some additional comments to Kconfig changes.

File: doc/nrf/ug_zigbee_configuring.rst [hunk 10]
```diff
@@ -19,12 +19,6 @@ Zigbee requires the following modules to properly operate in |NCS|:
 * :ref:`nrfxlib:zboss` available in nrfxlib, with the OSIF subsystem acting as the linking layer between the ZBOSS stack and |NCS|.
   OSIF implements a series of functions used by ZBOSS and is included in the |NCS|'s Zigbee subsystem.
   The files that handle the OSIF integration are located in :file:`nrf/subsys/zigbee/osif`.
-
-  * The ZBOSS stack library comes in production and development versions.
-    The production version is enabled by default with the :option:`CONFIG_ZIGBEE_LIBRARY_PRODUCTION` Kconfig option.
```
Comment (greg-fer) [inline doc/nrf/ug_zigbee_configuring.rst:24]: Will this ZBOSS Zigbee stack library be enabled by default after these changes are merged?
If yes, let's keep this line: ``This library is enabled by default with the :option:`name_of_option` Kconfig option.``

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: subsys/zigbee/Kconfig (hunk 3 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:196]: Fixed.

File: subsys/zigbee/Kconfig (hunk 4 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:194]: Fixed.

File: subsys/zigbee/Kconfig (hunk 5 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:164]: Fixed.

File: subsys/zigbee/Kconfig (hunk 6 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:162]: Fixed.

File: subsys/zigbee/Kconfig (hunk 7 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:173]: I agree. It is better to keep it consistent and change to FOTA.

File: subsys/zigbee/Kconfig (hunk 9 above)
Comment (tomchy) [reply subsys/zigbee/Kconfig:188]: Works for me

**Review by greg-fer — APPROVED** [review]

**Review by maciekfabia — APPROVED** [review]
LGTM

**Review by rlubos — APPROVED** [review]

**Pull request merged**